processing/src/main/java/org/apache/carbondata/processing/util/CarbonDataProcessorUtil.java

```diff
@@ -381,7 +381,9 @@ public static boolean isHeaderValid(String tableName, String[] csvHeader,
     List<CarbonDimension> dimensions =
         schema.getCarbonTable().getDimensionByTableName(factTableName);
     for (CarbonDimension dimension : dimensions) {
-      columnNames.add(dimension.getColName());
+      if (dimension.getSchemaOrdinal() != -1) {
```

Comment:
This is strange, why schemaOrdinal will equal to -1?

Reply:
Modified. Have defined a new property to identify a column as index column instead of relying on schemaOrdinal value.